app/Models/Ability.php

```diff
@@ -205,27 +205,29 @@ public function entryWithAttributes()
     }
 
     /**
-     * Attach an entity to the tag
+     * Attach an entity to the ability
      */
-    public function attachEntity(array $request): bool
+    public function attachEntity(array $request): int
     {
-        $entityId = Arr::get($request, 'entity_id');
-        $entity = Entity::with('abilities')->findOrFail($entityId);
-
-        // Make sure the tag isn't already attached to the entity
-        foreach ($entity->abilities as $ability) {
-            if ($ability->ability_id == $this->id) {
-                return true;
+        $entityIds = Arr::get($request, 'entities');
+        $entities = Entity::with('abilities')->findOrFail($entityIds);
+        $count = 0;
+        foreach ($entities as $entity) {
```

Comment:
Could we not replace all of this function with 
```php
$visibility = Arr::get($request, 'visibility_id', \App\Enums\Visibility::All);
$sync = [];
foreach ($entities as $entity) {
   $sync[$entity->id] = ['visibility_id' => $visibility];
}
$this->entities->sync($sync)
```
?

or does that lose information already saved on attached abilities like the "notes" and "position"?